### DISCUSSION TITLE
Tiling over large areas with stackstac

### DISCUSSION THREAD

**Discussion by ZZMitch** [gjoseph92/stackstac#245] [General]
Hello,

Sorry if this is not the right avenue for this discussion, but I wanted some feedback on best practice/ideas for using STAC/stackstac for wide area analysis. Some context...

I am interested in processing HLS satellite imagery across a large area that encompasses multiple UTM zones (e.g., Canada). I am testing means to split my area into reasonably sized tiles and processing each tile independently. To test, I created a fishnet grid of 120km squares in a national projection and extracted some HLS imagery using stackstac. 

In a perfect world, the output cube would have 4000 x 4000 pixels. However, because I need to provide a local UTM bounding box, the output is 4856 x 4856 pixels. This represent approximately 1.47x increase in pixels, which will overlap other fishnet squares and thus not be useful. 

Another option that I assume would avoid this would be use MGRS tiles directly, which are already in the local UTM. However that comes with its own set of redundancy issues - especially towards the poles. A nice visual is from this figure in [Bauer-Marschallinger 2023](https://www.sciencedirect.com/science/article/pii/S0924271623001971?via%3Dihub): 

![image](https://github.com/gjoseph92/stackstac/assets/78507308/15b6087a-3e4a-4711-9462-4637e01c965a)

Maybe there is some combination approach that would limit redundancy? For example, full squares in the center of a UTM zone that transition to slivers along the edge? But it seems like a lot of work to create a robust grid that meets all edge cases etc. 

I just wanted to make this post to see if others have worked through this type of issue and have some thoughts or ideas. Maybe I am missing something obvious!

**Comment by RichardScottOZ** [comment]
Take a look at some open data cube methods.

This and others in the base https://github.com/opendatacube/odc-geo

**Reply by ZZMitch** [reply]
Thanks I will check it out!
